base/repl/REPL.jl

```diff
@@ -853,7 +853,7 @@ function setup_interface(
                     end
                     # Check if input line starts with "julia> ", remove it if we are in prompt paste mode
                     jl_prompt_len = 7
-                    if (firstline || isprompt_paste) && (oldpos + jl_prompt_len <= sizeof(input) && input[oldpos:oldpos+jl_prompt_len-1] == JULIA_PROMPT)
+                    if (firstline || isprompt_paste) && (oldpos + jl_prompt_len <= sizeof(input) && input[oldpos:prevind(input, oldpos+jl_prompt_len)] == JULIA_PROMPT)
```

Comment:
A cleaner way of doing this check, which wouldn't involve playing with indices, would be `startswith(SubString(input, oldpos), JULIA_PROMPT)`.